horizon/facts/opal_forwarder.py

```diff
@@ -0,0 +1,61 @@
+from functools import cache
+from urllib.parse import urljoin
+
+from opal_common.fetcher.providers.http_fetch_provider import HttpFetcherConfig
+from opal_common.schemas.data import DataSourceEntry
+
+from config import sidecar_config
+from startup.remote_config import get_remote_config
+
+
+@cache
+def get_opal_data_base_url() -> str:
+    remote_config = get_remote_config()
+    org_id = remote_config.context.get("org_id")
+    proj_id = remote_config.context.get("project_id")
+    env_id = remote_config.context.get("env_id")
+    return urljoin(
+        sidecar_config.CONTROL_PLANE,
+        f"/v2/internal/opal_data/{org_id}/{proj_id}/{env_id}",
+    )
+
+
+@cache
+def get_opal_data_topic() -> str:
+    remote_config = get_remote_config()
+    pdp_client_id = remote_config.context.get("client_id")
+    topic = f"{pdp_client_id}:data:policy_data/{pdp_client_id}"
+    if sidecar_config.SHARD_ID:
+        topic += f"?shard_id={sidecar_config.SHARD_ID}"
+
+    return topic
+
+
+def generate_opal_data_source_entry(
+    obj_type: str,
+    obj_id: str,
+    obj_key: str,
```

Comment:
you have mypy error because you use body.get ( which can result with None ).
Do the None check either here or in the function that calls it